LayoutTests/fast/mediastream/mediastream-gc.html

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE HTML>
+<html>
+<body>
+<script src="../../resources/gc.js"></script>
+<script src="../../resources/testharness.js"></script>
+<script src="../../resources/testharnessreport.js"></script>
+<video id="video" autoplay=""></video>
+<script>
+async function doCaptureCheck(callback)
+{
+    if (!window.internals)
+        return;
+
+    let cptr = 0;
+    do {
+        await new Promise(resolve => setTimeout(resolve, 50));
+    } while (++cptr < 100 && !internals.pageMediaState().includes("HasActiveAudioCaptureDevice"));
+
+    assert_true(internals.pageMediaState().includes("HasActiveAudioCaptureDevice"));
+    assert_true(internals.pageMediaState().includes("HasActiveVideoCaptureDevice"));
+
+    if (callback)
+        callback();
+
+    cptr = 0;
+    do {
+        gc();
+        await new Promise(resolve => setTimeout(resolve, 50));
+    } while (++cptr < 100 && internals.pageMediaState().includes("HasActiveAudioCaptureDevice"));
+
+    assert_less_than(cptr, 100);
+    assert_false(internals.pageMediaState().includes("HasActiveAudioCaptureDevice"));
+    assert_false(internals.pageMediaState().includes("HasActiveVideoCaptureDevice"));
+}
+
+promise_test(() => {
+    navigator.mediaDevices.getUserMedia({audio:true, video : true});
+    return doCaptureCheck();
+}, "Media streams can be gced if they are not expected to fire events - 1");
+
+promise_test(async () => {
+    video.srcObject = await navigator.mediaDevices.getUserMedia({audio:true, video : true});
```

Comment:
This test is going to be flaky because we have a conservative GC.
Retry 5-10 times and check that at least one of them gets collected instead.